src/main/java/co/elastic/logstash/filters/elasticintegration/EventProcessorBuilder.java

```diff
@@ -308,17 +311,33 @@ public synchronized EventProcessor build(final PluginContext pluginContext) {
     }
 
     private static ScriptService initScriptService(final Settings settings, final ThreadPool threadPool) {
-        final List<Whitelist> painlessBaseWhitelist = getPainlessBaseWhiteList();
```

Comment:
review note: by using `PainlessPlugin#getScriptEngine(Settings, Set<ScriptContext>)` instead of `new PainlessScriptEngine(Settings, Map<ScriptContext<?>, List<Whitelist>>), we no longer need to manage the whole whitelists ourselves.